libs/s25main/world/MapBase.h

```diff
@@ -52,6 +52,8 @@ class MapBase
     MapPoint GetNeighbour2(MapPoint, unsigned dir) const;
     // Convenience functions for the above function
     MapCoord GetXA(MapPoint pt, Direction dir) const;
+    // Gets all neighbors (in all directions) for given position
+    std::vector<MapPoint> GetNeighbours(const MapPoint& pt) const;
```

Comment:
So this is GetPointsInRadius(pt, 1)?

Reply:
Yes, I tried that as well. But again, performance was much worse and I used it quite a few times looping over the whole map. Also tried to pre-calculate neighbors once for the whole map. But then the memory access seems to slow things down.

Reply:
Then maybe we should use your method in other places as well. Just checked it, looks indeed good and performant